### PR TITLE
Add guidance on writing tests for Linux+Windows

### PR DESCRIPTION
**Pull request by jjerger**
Working to set a convention for repo consistency and maintainability

### CODE REVIEW

File: imagetest/README.md [hunk 1]
```diff
@@ -72,6 +72,37 @@ Google Compute Engine startup script runner. The test files must specify a
...
+
+func TestConditionWindows() {
+      //Test something in Windows
+}
+func TestCondition() {
+      //Test something in Linux
+}
+
+func MyDefaultTest(
+      if runtime.GOOS == "windows" {
```
Comment (drewhli) [inline imagetest/README.md:94]: Nit: Tab space

File: imagetest/README.md [hunk 2]
```diff
@@ -72,6 +72,37 @@ Google Compute Engine startup script runner. The test files must specify a
...
+achieve similar results:
+
+```go
+import runtime
+
+func TestConditionWindows() {
+      //Test something in Windows
+}
+func TestCondition() {
+      //Test something in Linux
```
Comment (drewhli) [inline imagetest/README.md:90]: Nit: Tab space

File: imagetest/README.md [hunk 3]
```diff
@@ -72,6 +72,37 @@ Google Compute Engine startup script runner. The test files must specify a
...
+test definitions easier to read and maintain.
+
+For example, if the test MyDefaultTest() needs to run different commands to
+achieve similar results:
+
+```go
+import runtime
+
+func TestConditionWindows() {
+      //Test something in Windows
```
Comment (drewhli) [inline imagetest/README.md:87]: Nit: Tab space

**Review by drewhli — CHANGES_REQUESTED** [review]

File: imagetest/README.md [hunk 4]
```diff
@@ -72,6 +72,37 @@ Google Compute Engine startup script runner. The test files must specify a
...
+}
+func TestCondition() {
+      //Test something in Linux
+}
+
+func MyDefaultTest(
```
Comment (drewhli) [inline imagetest/README.md:93]: To make this more accurately reflect current tests, this should be
```go
func MyDefaultTest(t *testing.T) { ...
```

File: imagetest/README.md [hunk 5]
```diff
@@ -72,6 +72,37 @@ Google Compute Engine startup script runner. The test files must specify a
...
+      if runtime.GOOS == "windows" {
+		TestConditionWindows()
+	} else {
+		TestCondition()
+	}
+)
```
Comment (drewhli) [inline imagetest/README.md:99]: Nit: `}` instead of `)`

File: imagetest/README.md [hunk 6]
```diff
@@ -72,6 +72,37 @@ Google Compute Engine startup script runner. The test files must specify a
...
+When writing tests to run against both Linux and Windows, it is preferred to
+use separate functions within the same test where appropriate based on
+differences between OSes (ex. powershell vs bash commands). This makes the
+test definitions easier to read and maintain.
+
+For example, if the test MyDefaultTest() needs to run different commands to
+achieve similar results:
+
+```go
+import runtime
```
Comment (drewhli) [inline imagetest/README.md:84]: Add `testing` import.

**Review by drewhli — CHANGES_REQUESTED** [review]

File: imagetest/README.md (hunk 1 above)
Comment (jjerger) [reply imagetest/README.md:94]: Strange; Visual studio is set to tab 4 spaces yet it goes to 5 for markdown. Either way, fixing these.

File: imagetest/README.md (hunk 5 above)
Comment (jjerger) [reply imagetest/README.md:99]: Oops thats definitely a syntax error, not a nit. Thanks for the catch.

**Pull request merged**